pkg/security/podsecuritypolicy/types.go

```diff
@@ -32,12 +32,10 @@ import (
 // Provider provides the implementation to generate a new security
 // context based on constraints or validate an existing security context against constraints.
 type Provider interface {
-	// Create a PodSecurityContext based on the given constraints. Also returns an updated set
-	// of Pod annotations for alpha feature support.
-	CreatePodSecurityContext(pod *api.Pod) (*api.PodSecurityContext, map[string]string, error)
-	// Create a container SecurityContext based on the given constraints. Also returns an updated set
-	// of Pod annotations for alpha feature support.
-	CreateContainerSecurityContext(pod *api.Pod, container *api.Container) (*api.SecurityContext, map[string]string, error)
+	// Create a PodSecurityContext based on the given constraints.
+	CreatePodSecurityContext(pod *api.Pod) (*api.PodSecurityContext, error)
+	// Create a container SecurityContext based on the given constraints.
```

Comment:
nit: add a note that the pod is modified.

It's a little confusing to have the pod be modified, but also return a context that is then expected to be added to the pod... perhaps this API should just be changed to modify the pod directly?

Reply:
Even returning the security context is no longer necessary. I agree… the create... functions should be renamed to default…, add modify the pod and container directly, but a follow up  to do that is OK with me

Reply:
Thanks for the feedback! Updated. PTAL.